[Log Form] Added placeholder cache type to signature and log templates

bruzie opened the pull request:
Added support for the #GCType# placeholder in log and message templates. Only for the new logging experience (I believe the old logging page is not accessible any more).

Identifies cache type from the cache type icon using whatever selector is available for the page it is on.

2Abendsegler commented:
Thanks for your contribution!

The placeholders for log, signature, mail and message templates are a complex topic, so not necessarily suitable for starting. It is also always better to ask in advance whether we want to integrate an extension or not before putting in the effort.

There are a few things to consider with the placeholders.

The old log form can still be used if it is a draft and the parameter [Use old-fashioned log form to log a draft](https://www.geocaching.com/my/#GClhShowConfig#a#settings_drafts_old_log_form) is activated. The adjustments should also be made there. So in `function getGcTbUserInfo` at `// Log post.`.
It seems that this area is no longer accessible because `href.match(/\/cache_details\.aspx\?guid=/)` probably no longer works.

In the `// Log view`. area. in `function getGcTbUserInfo`, where you made an adjustment, the coding also no longer seems to be accessible.

In the area `// Log post cache new log page.` in `function getGcTbUserInfo`, the coding also no longer seems to be accessible.

And we should check whether the location we want to access is available. We should at least do this if it has already been done this way before. If we don't pay attention to this, an error message can appear if the location is no longer available. In `function getGcTbUserInfo`, for example, before using `$('.loggable-header > span > svg > title')[0].innerHTML`, it should be checked whether it is available.

My suggestion:

- I don't know whether we actually need the type in the mail and message templates. Maybe we shouldn't implement this point.
- The adjustment to the log templates seems sensible to me.
- The adjustment to the log signatures also seems sensible to me.

Maybe you can rework it accordingly and test everything again?

bruzie commented:
- Removed the Cache Type placeholder support for mail templates (both code and the help text)
- Added support for signatures on old log form

![Screenshot 2025-01-16 132236](https://github.com/user-attachments/assets/f162df26-6faf-4c10-b9dc-0626889a994c)
![Screenshot 2025-01-16 132150](https://github.com/user-attachments/assets/2229cbed-83eb-40b7-8d9b-c999a1dcbc18)

bruzie commented:
@2Abendsegler, I've checked in the changes, is there anything else I need to do with this pull request? (e.g. unassign myself)

2Abendsegler commented:
No, you don't need to do anything else.
Sorry, I'm so busy right now. I haven't had a chance to check yet.

2Abendsegler reviewed (approved):
Great, thanks for your development! 
I will merge your development.

After merging I will adjust two small things:
1. There is an error in TB Logs because the GC Type is not available there.
2. I have removed the GC Type for the "Possible placeholders" in TB Signatures.
